src/Payments.sol

```diff
@@ -1754,6 +1783,39 @@ contract Payments is ReentrancyGuard {
 
         return (fundedUntilEpoch, currentFunds, availableFunds, currentLockupRate);
     }
+
+    function burnFILForFees(IERC20 token, address recipient, uint256 requested) external payable {
+        Account storage fees = accounts[token][address(this)];
+        uint256 available = fees.funds;
+        require(available >= requested, Errors.WithdrawAmountExceedsAccumulatedFees(token, available, requested));
+
+        // TODO interval dutch auction
+        AuctionInfo storage auction = auctionInfo[token];
+        uint256 auctionPrice = uint256(auction.startPrice).decay(block.timestamp - auction.startTime);
+        require(msg.value >= auctionPrice, Errors.InsufficientNativeTokenForBurn(msg.value, auctionPrice));
+
+        auction.startPrice = uint88(auctionPrice * Dutch.RESET_FACTOR);
+        auction.startTime = uint168(block.timestamp);
+
+        fees.funds = available - requested;
+
+        (bool success,) = BURN_ADDRESS.call{value: msg.value}("");
+        require(success, Errors.NativeTransferFailed(BURN_ADDRESS, msg.value));
+
+        {
+            uint256 balanceBefore = token.balanceOf(address(this));
+            token.safeTransfer(recipient, requested);
```

Comment:
I had the idea to do an internal balance transfer instead, but we would not want these market makers to have to perform settlement check to withdraw.

Reply:
IMO (as a now outsider) we really should do this as an internal transfer. Market makers are already calling smart contract methods, they can handle a withdraw call and it makes it easier to reason about the security here.

Reply:
`withdraw` does `settleAccountLockupBeforeAndAfter` which does 2x `performSettlementCheck` etc. I don't think these make sense when they probably are neither a payer nor a payee. They probably will want to withdraw in the same transaction and we should make that cheap for them since the gas cost of completing this auction effectively reduces the FIL burn.

Reply:
1. Those calls will cost nothing (or almost nothing) if there are no rails, etc.
2. I expect anyone buying tokens here will be making multiple purchases over time.

You can implement this however you want, but it'll be a lot easier to reason about if there's exactly one way to deposit and withdraw funds.

If you do keep this as-is, I'd apply the global re-entrancy guard just to be safe and to avoid having to reason about any of this.

Reply:
I really think it's unnecessary and adds a lot of overhead that would reduce the FIL burn.

> very clean

I suspect not. We don't have other code for internal transfers so there's also complexity there. An internal transfer would probably need to do the same lockup operations as deposit.

>  writes a slot, emits an event

This happens twice: once for the native token (FIL), and once for the purchased token.

> a handful of slot reads

This adds deposit native and withdraw token to the operation. Here are the unnecessary reads that adds, assuming they have lockupRate=0:
* 4: funds and currentLockup for token and native 
* 1: token lockupFullySettled
* 2: native and token lockupLastSettledAt